src/App/Fossa/Analyze/Record.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.Analyze.Record
+  ( AnalyzeJournal (..),
+    AnalyzeEffects (..),
+    saveReplayLog,
+    loadReplayLog,
+  )
+where
+
+import App.Version (fullVersionDescription)
+import Control.Effect.Record
+import Data.Aeson
+import Data.Text (Text)
+import System.Directory (getCurrentDirectory)
+import System.Environment (getArgs)
+
+data AnalyzeJournal = AnalyzeJournal
+  { analyzeCommit :: Text,
+    analyzeEffects :: AnalyzeEffects,
+    analyzeArgs :: [String],
+    analyzeWorkdir :: FilePath
+  }
+  deriving (Eq, Ord, Show)
+
+data AnalyzeEffects = AnalyzeEffects
+  { effectsReadFS :: Journal,
+    effectsExec :: Journal
+  }
+  deriving (Eq, Ord, Show)
+
+instance FromJSON AnalyzeJournal where
+  parseJSON = withObject "AnalyzeJournal" $ \obj ->
+    AnalyzeJournal <$> obj .: "commit"
+      <*> obj .: "effects"
+      <*> obj .: "args"
+      <*> obj .: "workdir"
+
+instance FromJSON AnalyzeEffects where
+  parseJSON = withObject "AnalyzeEffects" $ \obj ->
+    AnalyzeEffects <$> obj .: "ReadFS"
+      <*> obj .: "Exec"
+
+instance ToJSON AnalyzeJournal where
+  toJSON AnalyzeJournal {..} =
+    object
+      [ "commit" .= analyzeCommit,
+        "args" .= analyzeArgs,
+        "workdir" .= analyzeWorkdir,
+        "effects" .= analyzeEffects
+      ]
+
+instance ToJSON AnalyzeEffects where
+  toJSON AnalyzeEffects {..} =
+    object
+      [ "ReadFS" .= effectsReadFS,
+        "Exec" .= effectsExec
+      ]
+
+saveReplayLog :: Journal -> Journal -> FilePath -> IO ()
+saveReplayLog readFSJournal execJournal path = do
```

Comment:
[question]: Is it worth making these more type-safe so we don't mix them up?  Either newtypes or something that looks like `Journal eff` where `eff` is some first-order effect?

Reply:
Oh yeah, good idea. a phantom type on `Journal` would work great here